pkg/nmstate/policy.go

```diff
@@ -0,0 +1,298 @@
+// with the NodeNetworkConfigurationPolicy definition in builder.
+func (builder *PolicyBuilder) Update(force bool) (*PolicyBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating the NodeNetworkConfigurationPolicy object %s",
+		builder.Definition.Name,
+	)
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+
+	if err != nil {
+		if force {
+			glog.V(100).Infof(
+				"Failed to update the NodeNetworkConfigurationPolicy object %s. "+
+					"Note: Force flag set, executed delete/create methods instead",
+				builder.Definition.Name,
+			)
+
+			builder, err := builder.Delete()
+
+			if err != nil {
+				glog.V(100).Infof(
+					"Failed to update the NodeNetworkConfigurationPolicy object %s, "+
+						"due to error in delete function",
+					builder.Definition.Name,
+				)
+
+				return nil, err
+			}
+
+			return builder.Create()
+		}
+	}
+
+	return builder, err
+}
+
+// WithInterfaceAndVFs adds SR-IOV VF configuration to the NodeNetworkConfigurationPolicy.
+func (builder *PolicyBuilder) WithInterfaceAndVFs(sriovInterface string, numberOfVF int) *PolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating NodeNetworkConfigurationPolicy %s with SR-IOV VF configuration: %d",
+		builder.Definition.Name, numberOfVF)
+
+	if sriovInterface == "" {
+		glog.V(100).Infof("The sriovInterface  can not be empty string")
+
+		builder.errorMsg = "The sriovInterface is empty sting"
+
+		return builder
+	}
+
+	nmStateDesiredStateInterfacesWithVfs := &DesiredState{
+		Interfaces: []Interface{
+			{
+				Name:  sriovInterface,
+				Type:  "ethernet",
+				State: "up",
+				Ethernet: Ethernet{
+					Sriov: Sriov{TotalVfs: numberOfVF},
+				},
+			},
+		},
+	}
+
+	nmStateInterfaceWithVfYaml, err := yaml.Marshal(nmStateDesiredStateInterfacesWithVfs)
+	if err != nil {
+		builder.errorMsg = "failed to Marshal NMState interface with VF"
+	}
+
+	builder.Definition.Spec.DesiredState = nmstateShared.NewState(string(nmStateInterfaceWithVfYaml))
+
+	return builder
+}
+
+// WaitUntilCondition waits for the duration of the defined timeout or until the
+// NodeNetworkConfigurationPolicy gets to a specific condition.
+func (builder *PolicyBuilder) WaitUntilCondition(condition nmstateShared.ConditionType, timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Waiting for the defined period until NodeNetworkConfigurationPolicy %s has condition %v",
+		builder.Definition.Name, condition)
+
+	if !builder.Exists() {
+		return fmt.Errorf("cannot wait for NodeNetworkConfigurationPolicy condition because it does not exist")
+	}
+
+	// Polls every retryInterval to determine if NodeNetworkConfigurationPolicy is in desired condition.
+	var err error
+
+	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
```

Comment:
Please set default re-try time once in 3 second. I would reccomend to define defaultRetryTime common var

Reply:
added